### DISCUSSION TITLE
How do I change URL in AudioPlayer

### DISCUSSION THREAD

**Discussion by Ephesian** [pschatzmann/arduino-audio-tools#1788] [Q&A] [answered]
I have created a sketch which uses the AudioPlayer class and I use the AudioSourceDynamicURL class as my source class.

I have created a webpage so that I can send new URLs to my player which is using a AI Thinker AudioKit with an ES8388 chip.

All is working fine when I declare a urls array and I specify the radios urls, the problem comes when I send the new url to the player I can not get it to work, and I am unable to find an example.  I have studied the class declarations and have tried various options but to no available.

Here is my code

```c
const char *ssid = "";
const char *pwd = "";
const char *urls[] = {
    "http://192.111.140.11:8272/stream"
    "http://31.132.6.142:4231/listen.pls?sid=1"
    };

  ICYStream urlStream(ssid,pwd);
  AudioSourceDynamicURL source(urlStream, urls, "audio/mp3");
  AudioBoardStream out(AudioKitEs8388V1);
  MP3DecoderHelix dec;
  AudioPlayer player(source, out, dec);
```

The code from setup() 
```c
    curSSID = WiFi.SSID().c_str();
    curPWD = WiFi.psk().c_str();
    
    urlStream.setSSID(curSSID);                       // TODO - needs fixing, but works if explicitly defined
    urlStream.setPassword(curPWD);
    auto config = out.defaultConfig();
    out.begin(config);

    dec.begin();
    player.setVolume(ini_block.reqvol);                                   // ini Volume from nvs
    //player.setMetadataCallback(showStreamTitle);             // The metadata
    player.begin();
```

This is my connect to host code which I think is where my problem is.
```c
  source.addURL(host.c_str());                                            // New url  
  source.selectStream("http://lynx.prostreaming.net:8272/stream");  // Hard coded url for testing
  player.setAudioSource(source);
  player.begin();
```

The code in loop()
```c
player.copy();
.... // other code
```

Any help would be welcome

**Comment by pschatzmann** [comment, marked as answer]
Why don't just just look at the class documentation of AudioSourceDynamicURL.
I don't understand why you are asking us to look up the method name, when you can do this yourself

**Reply by Ephesian** [reply]
Hello.  I appreciate your reply.  After searching through all the methods I discovered that it was due to URL.h, I found I was passing an url without the protocol (e.g. http://) and this was causing  a return without parsing the url line 62 in url.h

The easy fix which I applied was to do a quick check on my url string and add http if it was missing (assuming http) and then update selectStream.  This has resolved the issue.

I also replaced player.begin() with player.play(), as player.begin() was over riding the url with index 0 of the array.
